src/components/menu/menu-types.ts

```diff
@@ -93,6 +93,24 @@ class MenuRevealType extends MenuType {
 }
 MenuController.registerType('reveal', MenuRevealType);
 
+/**
+ * @hidden
+ * Menu Split Type
+ * The content slides over to reveal the menu underneath and shrinks so all the content remains inside the viewport.
+ * The menu itself, which is under the content, does not move.
+ */
+class MenuSplitType extends MenuType {
+    constructor(menu: Menu, plt: Platform) {
+        super(plt);
+
+        const openedX = (menu.width() * (menu.isRightSide ? -1 : 1));
+        const contentOpen = new Animation(plt, menu.getContentElement());
+        contentOpen.fromTo('translateX', '0px', openedX + 'px');
+        contentOpen.fromTo('width', '100%', `calc(100% - ${Math.abs(openedX)}px)`);
```

Comment:
these 2 lines (and line 106) can be joined to:
`contentOpen.fromTo(menu.isRightSide ? 'right' : 'left', '0px', menu.width()+ 'px');`

So it won't look weird when it changes the width